src/profile-logic/profile-tree.js

```diff
@@ -187,101 +191,171 @@ class ProfileTree {
 
 export type ProfileTreeClass = ProfileTree;
 
-export function getCallTree(
+function _getInvertedStackSelfTimes(
```

Comment:
I split out the timing code into separate functions because I found them confusing. I also wrote some tests covering timing calculations.